go/cmd/dolt/commands/sqlserver/server.go

```diff
@@ -261,12 +278,47 @@ func getDbStates(ctx context.Context, mrEnv env.MultiRepoEnv, dbs []dsqle.Databa
 		}
 
 		head := dEnv.RepoStateReader().CWBHeadSpec()
-		headCommit, err := dEnv.DoltDB.Resolve(ctx, head, dEnv.RepoStateReader().CWBHeadRef())
+
+		var ws *doltdb.WorkingSet
+		var headCommit *doltdb.Commit
+		var err error
+		if _, val, ok := sql.SystemVariables.GetGlobal(DoltDefaultBranchKey); ok {
+			if refStr, ok := val.(string); ok && ref.IsRef(refStr) {
+				headRef, err := ref.Parse(refStr)
+				if err != nil {
+					break
+				}
+				headCommit, err = dEnv.DoltDB.Resolve(ctx, head, headRef)
+				if err != nil {
+					return nil, err
+				}
+
+				workingSetRef, err := ref.WorkingSetRefForHead(headRef)
+				if err != nil {
+					return nil, err
+				}
+
+				ws, err = dEnv.DoltDB.ResolveWorkingSet(ctx, workingSetRef)
+				if err != nil {
+					return nil, err
+				}
+
+				dbStates = append(dbStates, dsess.InitialDbState{
+					Db:         db,
+					HeadCommit: headCommit,
+					WorkingSet: ws,
+					DbData:     dEnv.DbData(),
+				})
+				continue
```

Comment:
The continues are getting a little out of hand, making this method hard to read.

You should rewrite this to be simple conditional logic that gets a db state one of 3 ways

```go
if (denv == nil) {
   dbState = getRevisionDbState(...)
} else if  (_, val, ok := sql.SystemVariables.GetGlobal(DoltDefaultBranchKey); ok) {
  dbState = getDbStateForDefaultBranch(...)
} else {
  dbstate = getDbStateForDEnv(...)
} 

dbStates.append...
```